base/path.jl

```diff
@@ -57,7 +55,14 @@ splitdrive(path::AbstractString)
 
 Return the current user's home directory.
 """
-homedir()
+function homedir()
+    buf = Vector{Cchar}(1024)
+    sz = Ref{Csize_t}(sizeof(buf))
+    rc = ccall(:uv_os_homedir, Cint, (Ptr{Cchar}, Ptr{Csize_t}), buf, sz)
+    rc == 0 || error("unable to retrieve home directory")
```

Comment:
Hard to imagine it will happen but the correct way is to use `sz[] - 1` to resize the array.